### ISSUE TITLE
Include all docstrings in the manual

### ISSUE THREAD

**Issue by MichaelHatherly** [JuliaLang/julia#19529]
Currently there are a significant number (200+) docstrings in `Base` and it's submodules
that are not included in the manual anywhere. To see the current list of docstrings that
still need to be included change the following line in `doc/make.jl`

```julia
    checkdocs = :none,
```

to

```julia
    checkdocs = :all,
```

and build the docs with

```
make docs
```

from the root directory.

There's some docstrings for names that are exported and should definitely find there way into the manual somewhere, as well as quite a few internal docstrings that probably don't need to be included or could do with some improvements before inclusion.

(I'm marking this issue as up-for-grabs since it can readily be distributed between many contributors.)

**Comment by tkelman** [comment]
And these weren't included in the rst manual either?

**Comment by MichaelHatherly** [comment]
Yes, none of them were ever included in the manual, so this would just be adding extra docstrings to the manual.

**Comment by tkelman** [comment]
I'm seeing several html files missing from gh-pages that were present in the previous html output:
```
devdocs/C.html
devdocs/julia.html
stdlib/stacktraces.html
```

**Comment by MichaelHatherly** [comment]
Source files still exist as far as I can see, just weren't included in `makedocs`. `C.md` and `julia.md` are just "index" pages, which just duplicate what's already in the side menu, so they can probably just be removed. I'll add back `stacktraces.md` though.

**Comment by mortenpi** [comment]
The current state is that we have [195-ish docstrings attached to exported names that are not spliced into the manual](https://gist.github.com/mortenpi/7bac45cfafcae4175750febdfbeb4aec#file-missing-docstrings-txt) (and [581 total](https://gist.github.com/mortenpi/7bac45cfafcae4175750febdfbeb4aec#file-missing-docstrings-all-txt)). I say "-ish" because some of the ones on the list are just exports from internal modules and not part of the public API.

- [ ] All the missing docstrings of exports should be added to their respective manual pages.
- [ ] Check this on CI. Documenter's `checkdocs = :exports` almost does this, but does get confused by the exports from internal modules. It should probably just check for exports from the top-level modules.

**Comment by IanButterworth** [comment]
The number of `checkdocs = :exports` is up to 298 now. https://github.com/JuliaLang/julia/pull/55267#issuecomment-2253000508

**Comment by IanButterworth** [comment]
I've renamed this issue to be about public docstrings, as I think that's the correct requirement.

Requires a Documenter.jl feature https://github.com/JuliaDocs/Documenter.jl/issues/1506

**Comment by nsajko** [comment]
* `Base.donotdelete` does not appear in the manual

    * Made public in PR #55774

* `Base.blackbox` does not appear in the manual

    * Added and made public in PR #61438
